lib/modules.nix

```diff
@@ -58,10 +58,7 @@ rec {
 
       closed = closeModules (modules ++ [ internalModule ]) ({ inherit config options; lib = import ./.; } // specialArgs);
 
-      # Note: the list of modules is reversed to maintain backward
-      # compatibility with the old module system.  Not sure if this is
-      # the most sensible policy.
-      options = mergeModules prefix (reverseList closed);
+      options = mergeModules prefix (filterModules (specialArgs.modulesPath or "") closed);
```

Comment:
nit: This is adding a meaning to a name which used to be ignored by the module system.  Add a paragraph to the existing `specialArgs` comment to describe what is the intended meaning of `modulesPath`.